src/app/features/public-form/edit-prefilled-values/edit-prefilled-values.component.ts

```diff
@@ -110,11 +118,24 @@ export class EditPrefilledValuesComponent
     }
 
     const updatedFields: FormFieldConfig[] = value.prefilledValue.map(
-      ({ field, defaultValue, hideFromForm }) => ({
-        id: field,
-        defaultValue,
-        hideFromForm: hideFromForm ?? true,
-      }),
+      ({ field, defaultValue, hideFromForm }) => {
+        const transformedValue = this.entitySchemaService.valueToDatabaseFormat(
+          defaultValue.config?.value,
+          this.getSchemaField(field),
+        );
```

Comment:
I think we should try to move this into the admin-default-value-static component. The defaultValues should always be stored in database format, so that component could make sure it only sets/emits it in the correctly transformed format to the outside already